### ISSUE TITLE
WPS115 (Found upper-case constant in a class) in enum.Enum.

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#750]
# Bug report

## What's wrong

WPS115 violation forbids uppercase attributes for classes.

## How is that should be

However, [according to official documentation](https://docs.python.org/3/library/enum.html#creating-an-enum), `enum.Enum` attributes should be uppercased.

BTW, checker for `enum.Enum` attributes case can be a separate violation.

**Comment by orsinium** [comment]
It's not only about style.

> Enum members are instances of their Enum class, and are normally accessed as EnumClass.member. Under certain circumstances they can also be accessed as EnumClass.member.member, but you should never do this as that lookup may fail or, worse, return something besides the Enum member you are looking for (this is another good reason to use all-uppercase names for members):

**Comment by sobolevn** [comment]
Hm, I was actually using `snake_case` for `enums` for my whole life.

**Comment by orsinium** [comment]
`¯\_(ツ)_/¯`

I think official Python documentation has value here. Enum isn't a regular class.

**Comment by sobolevn** [comment]
I agree. Let's do it!

**Comment by sobolevn** [comment]
Ok, I have tried several solutions, but we cannot statically tell that this class is enum or not.
We can only check for `enum.Enum` or `Enum` (and subtypes) in class bases. But, that's not enough. We cannot find custom `Enum` subtypes, aliases (`import enum as en`), etc.

So, we cannot make this rule reliable. The our philosophy - that's not how we define rules.

I have updated docs, that we will continue to use `snake_case` for enums.
